src/object.c

```diff
@@ -907,7 +907,7 @@ int paint_large_scenery(int flags, int ebx, int ecx, int edx, rct_drawpixelinfo*
 			chunk += 1038;
 		}
 
-		scenery_type->large_scenery.var_0C = (uint32)chunk;
+		scenery_type->large_scenery.tiles = (rct_large_scenery_tile*)chunk;
 
 		while (*((uint16*)chunk) != 0xFFFF){
 			chunk += 9;
```

Comment:
could change this to `+= sizeof(rct_large_scenery_tile);` and add a comment saying skip over scenery tiles.